src/ios/CDVNavitiaSDKUX.swift

```diff
@@ -0,0 +1,140 @@
+            let storyboard: UIStoryboard = UIStoryboard(name: "Journey", bundle: bundle)
+            
+            let rootViewController: JourneySolutionsController = storyboard.instantiateInitialViewController() as! JourneySolutionsController
+            rootViewController.setProps(with: params)
+            
+            let navigationController: UINavigationController = UINavigationController(rootViewController: rootViewController)
+            self.viewController.present(navigationController, animated: true, completion: nil)
+            
+            pluginResult = CDVPluginResult(status: CDVCommandStatus_OK)
+        } else {
+            pluginResult = CDVPluginResult(status: CDVCommandStatus_ERROR, messageAs: "No parameter specified in invokeJourneyResults")
+        }
+        
+        self.commandDelegate!.send(pluginResult, callbackId: command.callbackId)
+    }
+
+    func getJourneyInParameters(from arguments: [String: Any]) -> JourneySolutionsController.InParameters {
+        let originId: String = arguments["originId"] as? String ?? ""
+        let destinationId: String = arguments["destinationId"] as? String ?? ""
+
+        var params: JourneySolutionsController.InParameters = JourneySolutionsController.InParameters(originId: originId, destinationId: destinationId)
+
+        if (arguments["originLabel"] != nil) {
+            params.originLabel = arguments["originLabel"] as? String ?? ""
+        }
+        if (arguments["destinationLabel"] != nil) {
+            params.destinationLabel = arguments["destinationLabel"] as? String ?? ""
+        }
+        if (arguments["datetimeRepresents"] != nil) {
+            params.datetimeRepresents = self.getDatetimeRepresents(from: arguments["datetimeRepresents"] as? String ?? "")
+        }
+        if (arguments["datetime"] != nil) {
+            params.datetime = getDatetime(from: arguments["datetime"] as? String ?? "")
+        }
+        if (arguments["forbiddenUris"] != nil) {
+            params.forbiddenUris = arguments["forbiddenUris"] as? [String] ?? []
+        }
+        if (arguments["firstSectionModes"] != nil) {
+            params.firstSectionModes = self.getFirstSectionModes(from: arguments["firstSectionModes"] as? [String] ?? [])
+        }
+        if (arguments["lastSectionModes"] != nil) {
+            params.lastSectionModes = self.getLastSectionModes(from: arguments["lastSectionModes"] as? [String] ?? [])
+        }
+        if (arguments["count"] != nil) {
+            params.count = arguments["count"] as? Int32 ?? 0
+        }
+        if (arguments["minNbJourneys"] != nil) {
+            params.minNbJourneys = arguments["minNbJourneys"] as? Int32 ?? 0
+        }
+        if (arguments["maxNbJourneys"] != nil) {
+            params.maxNbJourneys = arguments["maxNbJourneys"] as? Int32 ?? 0
+        }
+
+        return params
+    }
+
+    func getDatetimeRepresents(from argument: String) -> JourneysRequestBuilder.DatetimeRepresents {
```

Comment:
For string to enum can you use a more generic code ?

something like this maybe https://github.com/CanalTP/CDVNavitiaSDK/blob/master/src/ios/CDVNavitiaSDK.swift#L33